src/sentry/api/endpoints/organization_releases.py

```diff
@@ -161,6 +162,11 @@ def post(self, request, organization):
             if commit_list:
                 release.set_commits(commit_list)
 
+            head_commits = result.get('head_commits')
+            if head_commits:
+                fetch_commits = request.user.is_authenticated() and not commit_list
+                release.set_head_commits(head_commits, request.user, fetch_commits=fetch_commits)
```

Comment:
As a follow up, we're going to move this to a celery task, but for now, this is goign to be fine. A celery task will allow us to make multiple attempts and handle failures better. Also to not block the main http process while waiting on GitHub API.